File: database/access_database.js
const { MongoClient } = require('mongodb');

function novo_sentimento(sentimento) {
    const uri = "mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
    const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });
    client.connect(err => {
      const collection = client.db("test").collection("devices");
      key = "sentimentos."+sentimento;
      collection.updateOne(
          {},
          {
              "$inc": { key : 1 }
            });
      client.close();
    });
}

function novo_acesso_agora(){
    const uri = "mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
    const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });
    client.connect(err => {
      const collection = client.db("test").collection("devices");
      collection.updateOne(
          {},
          {
              "$push": { "acessos" : new Date()}
          });
      client.close();
    });
}

function nova_pesquisa(query){
    const uri = "mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority";
    const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });
    client.connect(err => {
      const collection = client.db("test").collection("devices");
      key = "pesquisas."+query;
      collection.updateOne(
          {},
          {
              "$inc": { key : 1 }
          });
      client.close();
    });
}

File: database/access_database.py
import pymongo

#Retorna uma visão do documento principal do bd em forma de dicionário
def get_view():
    Client = pymongo.MongoClient("mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority")
    colecao = Client.Listify_db.statistics
    return colecao.find_one()

#Atualiza o bd com a contagem de um novo sentimento
def set_novo_sentimento(sentimento):
    Client = pymongo.MongoClient("mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority")
    colecao = Client.Listify_db.statistics
    colecao.update_one({},{"$inc": {"sentimentos."+sentimento:1}})

#Atualiza o bd com a data de um novo acesso
def set_novo_acesso(date):
    Client = pymongo.MongoClient("mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority")
    colecao = Client.Listify_db.statistics
    colecao.update_one({},{"$push": {"acessos": date}})

#Atualiza o bd com a chave de uma nova pesquisa
def set_nova_pesquisa(query):
    Client = pymongo.MongoClient("mongodb+srv://ListifyES:musiquinhastopzera@projetoes1.tm97r.mongodb.net/myFirstDatabase?retryWrites=true&w=majority")
    colecao = Client.Listify_db.statistics
    colecao.update_one({},{"$inc": {"pesquisas."+query: 1}})

